app/controllers/concerns/batch_processable.rb

```diff
@@ -1,14 +1,8 @@
 module BatchProcessable
   extend ActiveSupport::Concern
 
-  def load_selected_objects(klass)
-    klass = klass.accessible_by(current_ability)
-    if params[:select_all].present?
-      klass.all.to_a
-    elsif params[:selected].present?
-      params[:selected].keys.collect{ |id| klass.find_by_id(id) }.compact
-    else
-      []
+  def load_selected_objects(rel)
+    return rel if params[:select_all].present? || params[:selected].nil?
+    rel.where(id: params[:selected].keys)
```

Comment:
I'm curious how this works with pagination if you do the "select all pages" thing. Wouldn't we have to _undo_ the pagination scope? This makes me wonder if this combination is fully tested...